### ISSUE TITLE
Exception marker interfaces shsould extend `Throwable`

### ISSUE THREAD

**Issue by Ocramius** [phar-io/version#16]
Following code is generally not valid from a static perspective (note: still works at runtime)

```php
interface MyMarker
{
}

try {
} catch (MyMarker $e) {
}
```

The rationale is that the contract to be followed for a symbol to be used in a `catch ()` is that it should implement `Throwable`.

In practice, the idea is that the interface should be extending `Throwable`:

```php
interface MyMarker extends Throwable
{
}
```

This currently affects following:

https://github.com/phar-io/version/blob/45a2ec53a73c70ce41d55cedef9063630abaf1b6/src/exceptions/Exception.php#L11-L14

**Comment by sebastianbergmann** [comment]
This can only be addressed after support for PHP 5.6 has been dropped (as there is no `\Throwable` interface in PHP 5.6).

**Comment by theseer** [comment]
Looks like I implemented this without properly annotating it in the changelog.

It's part of 3.0.0.
